enhancements/rhcos/split-rhcos-into-layers.md

```diff
@@ -0,0 +1,428 @@
+Firstly, because it's the one we'll start with, and secondly, because it's the one that
+will most impact OpenShift. The second part will require changes that are more contained
+to the CoreOS and ART teams. That said, we felt it important to introduce them together
+in this enhancement to provide a larger picture.
+
+Diving into the first part then, at a high-level this is split into three phases.
+
+#### Phase 1: Using RHEL/CentOS Stream-only bootimages
+
+In this phase, the goal is to have the OpenShift installer use bootimages containing only
+RHEL/CentOS Stream content.
+
+To do this, we will start building two new streams in the RHCOS pipeline containing only pure
+RHEL/CentOS Stream content (let's call these the "pure RHEL stream" and "pure CentOS stream").
+Those streams will also be building the usual bootimages and uploading them to cloud.
+
+Once we have these bootimages, we can better start adapting components that will need it
+to account for the lack of OpenShift components in the bootimages. Likely suspects here are
+any components involved in the bringup of the cluster (installer, Assisted Installer, MCO, etc...).
+
+Once all known issues have been resolved, the installer bootimage metadata will be updated
+to point to these new images. At this point, the OpenShift node image (i.e. the
+`rhel-coreos` image in the release payload) has not changed.
+
+Similarly, the installer bootimage metadata for OKD will be updated to use the new bootimages
+from the CentOS stream.
+
+#### Phase 2: Switching `rhel-coreos` in CI
+
+In this phase, the goal is to have the OpenShift node image be built in, and used by, CI.
+
+To do this, we will configure the RHCOS pipeline to push the bootable container image output
+from the pure RHEL stream to a development Quay repo. We will then create a new Prow CI job
+which will build the OpenShift node image by layering the OpenShift components on top: RPM
+packages, config files, etc...
+
+After some testing, the Prow job will be configured to push to the `rhel-coreos` tag in the CI
+imagestream. The ART job that syncs the RHCOS container images to CI will be disabled.
+
+(Ideally, we'd use a new tag for this here and in the release payload. E.g. `node` or `os`
+would be more appropriate, but... there's a lot of work involved in doing this.)
+
+#### Phase 3: Switching `rhel-coreos` in production
+
+In this phase, the goal is to have the OpenShift node image be built in, and used by, production.
+
+To do this, we will setup a Konflux pipeline to build the layered node image. After some
+additional testing, this layered image will replace the current `rhel-coreos` image in the
+production release payload.
+
+### Workflow Description
+
+This enhancement should have no immediate impact on cluster administrators. However, better
+alignment with image mode for RHEL should result in better documentation and tooling down the line.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+TODO
```

Comment:
One thing I've mentioned in other places here is that if we can get to the point of shrinking our use case for Ignition to be very small by using container images instead, then we don't require the MCS anymore, which would definitely simplify Hypershift.

Reply:
I'll add a note for that, though probably in the "Follow-ups" section instead.